core/lib/zksync_core/src/state_keeper/io/mod.rs

```diff
@@ -96,209 +136,107 @@ pub trait StateKeeperIO: 'static + Send + IoSealCriteria {
     /// Marks the transaction as "rejected", e.g. one that is not correct and can't be executed.
     async fn reject(&mut self, tx: &Transaction, error: &str) -> anyhow::Result<()>;
 
-    /// Marks the miniblock (aka L2 block) as sealed. Returns the timestamp for the next miniblock.
-    async fn seal_miniblock(&mut self, updates_manager: &UpdatesManager);
-    /// Marks the L1 batch as sealed.
-    async fn seal_l1_batch(
+    /// Loads base system contracts with the specified version.
+    async fn load_base_system_contracts(
         &mut self,
-        witness_block_state: Option<WitnessBlockState>,
-        updates_manager: UpdatesManager,
-        l1_batch_env: &L1BatchEnv,
-        finished_batch: FinishedL1Batch,
-    ) -> anyhow::Result<()>;
-
-    /// Loads protocol version of the previous l1 batch.
-    async fn load_previous_batch_version_id(&mut self) -> anyhow::Result<ProtocolVersionId>;
+        protocol_version: ProtocolVersionId,
+        cursor: &IoCursor,
+    ) -> anyhow::Result<BaseSystemContracts>;
+    /// Loads protocol version of the specified L1 batch, which is guaranteed to exist in the storage.
+    async fn load_batch_version_id(
+        &mut self,
+        number: L1BatchNumber,
+    ) -> anyhow::Result<ProtocolVersionId>;
     /// Loads protocol upgrade tx for given version.
     async fn load_upgrade_tx(
         &mut self,
         version_id: ProtocolVersionId,
     ) -> anyhow::Result<Option<ProtocolUpgradeTx>>;
 }
 
-impl fmt::Debug for dyn StateKeeperIO {
-    fn fmt(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
-        formatter
-            .debug_struct("StateKeeperIO")
-            .field("current_l1_batch_number", &self.current_l1_batch_number())
-            .field("current_miniblock_number", &self.current_miniblock_number())
-            .finish()
+/// Handler for state keeper outputs (miniblocks and L1 batches).
+#[async_trait]
+pub trait HandleStateKeeperOutput: 'static + Send + fmt::Debug {
```

Comment:
Nit: I remember that there is a suggestion to [use imperatives for traits](https://github.com/rust-lang/api-guidelines/discussions/28), but it isn't universally followed even in std (e.g. `Iterator`, `Future`, etc). 

To me, it feels like an implicit convention is as follows:
- If there is a single method that majorly describes the behavior of a trait, it should be used as the name of a trait (`clone`, `from`, `into`, etc).
- Otherwise, the trait is treated as an interface in a more general sense and should be named as you would name an interface in any other language (excluding language-specific rules, like the `I` prefix).

In this particular case, the naming seems to be artificially affected and not really natural.  IMHO, `StateKeeperOutputHandler` reads _much_ more naturally.

Not a strong opinion tho.